usr/sbin/pkcsstats/pkcsstats.c

```diff
@@ -406,24 +407,45 @@ int get_token_infos(CK_FUNCTION_LIST_PTR func_list, CK_SLOT_ID slot,
     return 0;
 }
 
+struct display_mech {
+    bool json;
+    bool first_mech;
+};
+
 static int display_mech_cb(CK_MECHANISM_TYPE mech, const char *mech_name,
                            CK_BYTE *mech_data, CK_ULONG mech_size,
                            CK_ULONG ofs, void *private)
 {
     counter_t *counter = (counter_t *)mech_data;
+    struct display_mech *dm = private;
     int i;
 
     UNUSED(mech);
-    UNUSED(private);
     UNUSED(ofs);
 
-    printf("%-30s |", mech_name);
+    if (dm->json && dm->first_mech == false)
+        printf(",");
+
+    if (dm->json)
+        printf("\n\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\"mechanism\": \"%s\",\n", mech_name);
```

Comment:
This is supposed to be the machine-readable output, right?  If yes, why do we care so much about indentation here?  We could just produce minimal JSON (maybe not minimal since I am not opposed to new lines) instead of this vast amount of tabs...

Reply:
Well, yes, but I also wanted it to be human readable to some extend....
I could indent with blanks (i.e. 2 blanks) instead of tabs, but that would produce more data than tabs....